File: RA1/JAVA/EjemploPokemon/Pokemon.java
package EjemploPokemon;

import java.util.Random;

public class Pokemon{
    //Atributos
    private String nombre;
    private String tipo;
    private int vida;
    private int exp;
    private int pc;
    private int puntosAtaque;
    private int puntosAtaqueCargado;
    private String estado;

    //Métodos
    public Pokemon(String nombre, String tipo, int pc){
        this.nombre = nombre;
        this.tipo = tipo;
        this.pc = pc;

        Random aleatorio = new Random(pc);
        //aleatorio.nextInt(m) + n; -> [n,m-1]
        this.puntosAtaque = aleatorio.nextInt(11) + 10; //nextInt(n) -> [0,n-1]
        this.puntosAtaqueCargado = aleatorio.nextInt(51) + 50;
    }

    public int getPc(){
        return this.pc;
    }

    public String toString(){
        return "\nEl pokemon se llama " + this.nombre + " de tipo " + this.tipo + " con PC " + this.pc + "\n - Ataque Simple = " + this.puntosAtaque + "\n - Ataque Cargado = " + this.puntosAtaqueCargado;
    }   
}

File: RA1/PYTHON/Pokemon/Pokemon.py
import random

class Pokemon:

    def __init__(self, nombre, tipo, pc):
        self.nombre = nombre
        self.tipo = tipo
        self.pc = pc
        self.vida = None

        self.puntos_ataque = random.randrange(10,21) #(3,8) -> [3,7]
        self.puntos_ataque_cargado = random.randrange(50,101)
        self.estado = None

    def get_pc(self):
        return self.pc

    def __str__(self):
        return "\nEl pokemon se llama " + self.nombre + " de tipo " + self.tipo + " con PC " + str(self.pc) + "\n - Ataque Simple = " + str(self.puntos_ataque) + "\n - Ataque Cargado = " + str(self.puntos_ataque_cargado)
        